src/openrct2/peep/Guest.cpp

```diff
@@ -5248,12 +5248,14 @@ void rct_peep::UpdateWalking()
         }
     }
 
-    rct_tile_element * path_element = map_get_path_element_at(destination_x / 32, destination_y / 32, next_z);
-    if (path_element && path_element->flags & TILE_ELEMENT_FLAG_BLOCKED_BY_VEHICLE)
+    // Check if vehicle is blocking the destination tile
+    auto curPos = TileCoordsXYZ(CoordsXYZ { x, y, z });
+    auto dstPos = TileCoordsXYZ(CoordsXY { destination_x, destination_y }, next_z);
+    if (curPos.x != dstPos.x || curPos.y != dstPos.y)
```

Comment:
We should probably overload the == operator for TileCoordsXYZ so we can compare them directly.